### DISCUSSION TITLE
Stance on frog-protocols

### DISCUSSION THREAD

**Discussion by tfemby** [labwc/labwc#2184] [General]
With the beginning of [frog-protocols](https://github.com/misyltoad/frog-protocols) I'm just wondering what the stance is on this repo. It's new right now but is this something that Labwc might consider adding in the future?

**Comment by johanmalm** [comment]
Good question :smile: Well, the simple answer is that it depends (see last bullet point)

Let me give you my opinion, but I want to caveat it by saying that it's just my view and within the `labwc` team we often don't agree on initial contact, which is fine btw and so far we've managed to discuss and move on with good outcomes.

As an Engineer, I think that initiatives like frog-protocols are both good and bad. Let me explain:

They are bad in the sense that they may cause fragmentation (yet another standard, etc), undermine w-p and likely lead to nugatory work with client/server implementations, regression, version clashes, packaging mess, etc. You get my drift.

They are good in that they may speed up overall implementation timescales and may serve a greater good because standards are often the result of chaotic, real-life evolution. In other words, if you implement (potentially less thought through) protocols in clients/servers you get 'real' feedback so that you can improve over time accepting that that processes is probably messier and is likely to break things along the way.

As a point of reference here - and at a quick glance - it looks like the frog-protocols repo is only five days old. Interestingly though it's already in the Arch Linux extra repo.

Now as for `labwc`:

1. I (could probably say 'we' here) generally take a conservative approach and tend not to be the first ones to jump in with 'new' and 'shiny' because we value stability within current scope which is hard enough to achieve. 
2. So saying 'yes' to a set of protocols under a new governance arrangement with no historic credibility, no obvious long-term commitment, no proven adoption rate, etc., seems a complete no-go. Now, just to clarify here, @misyltoad did contribute quite a bit to `labwc` in the early days and is in my mind very competent and a force for good in the wider Wayland eco-system on the whole, so I'm not doubting the technical goodness, necessity, etc. But hopefully my point is clear nonetheless.
3. There is of course a balance in these things which is why it's not right to take a 100% black and white stance. For example, I don't think compositors like sway, `labwc`, etc would have got much adoption without many of the original `wlr` protocols (e.g. the layer-shell one). But now that they are somewhat established, the move is to upstream them in w-p.
4. On the whole, `labwc` is feature complete in terms of what I initially set out to do. Now, my heart is increasingly in enabling Desktop Environments like LXQt, Raspberry Pi OS and XFCE; and if I were them I'd probably think that not everything is yet in place within w-p to support a basic desktop environment on par with the equivalent on X11, without some custom protocols or other custom IPC. A couple of quick examples would be IPC for workspaces and keyboard-layouts. On the workspace front we look to merge a cosmic variant of the proposed w-p workspace protocol (https://github.com/labwc/labwc/pull/2030). But just to be clear, this is because there is an obvious need, the w-p ones has been 'stuck' for a long time and has no hint of movement, and we purposefully don't want to use a draft revision of the w-p one because if/when it is approved we don't want a clash. And when an w-p one exists we're likely to deprecate the cosmic one, so I think of it as a temporary crutch.
5. I think that sustainability in software tools like `labwc` does in the long-run very much depend on carefully managing scope and working within limits. Sometimes it's really hard to feel that you're doing the right thing in saying 'no' because it's 'just one flag' or just one more thing, or just one work-around. But most people (in my view) who have been around the buoy a few times often regret letting too many new features, ill thought trough code, too much coupling, too many config options, too many work-arounds, etc in the repo. And re-writes often reduce scope. BUT having said all that, if people don't want to use the software tool (`labwc` in this case) because it doesn't support stuff that they feel they need, then that's obviously no good either. So for example, if people in the gaming community said that `labwc` is a good compositor for them, but that the lack of `frog-color-management` (or whatever) makes it unattractive and ideally if they're willing to contribute the code plus be there to support then that's a very different question. I suggest we leave the discussion until that point. And just to clarify, we're already in that position with tearing/vrr/etc so it's not a theoretical supposition to avoid having to answer, it's just the natural way of things in my view.

**Reply by tfemby** [reply]
Thanks Johan for the reply!

I was asking this mainly out of pure curiosity as frog-protocols seems to have Valve and a few distro's backing/blessing (as you noted). I do appreciate the conservative, "Wait and see," style stance; especially since frog's still in it's infancy. Not being totally against the idea of incorporating something if somebody submits an implementation is nice to hear. Labwc is still open to new ideas!

That being said, I'm pretty satisfied with this answer :)

**Reply by stefonarch** [reply]
> As a point of reference here - and at a quick glance - it looks like the frog-protocols repo is only five days old. Interestingly though it's already in the Arch Linux extra repo.

Could this be the reason?
https://twitter.com/SteamDB/status/1839946113152893298/photo/1

**Reply by tfemby** [reply]
Not totally sure but Fedora also has packages for it as well.
https://src.fedoraproject.org/rpms/frog-protocols

Valve employee Pierre-Loup Griffais also gave his take on it in the W-P MR.
https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/31329/#note_2580740

You may be correct in that it's because of shared goals with Arch. I have a feeling there's probably some more communication going on behind the scenes for things to get packages this fast.
